src/Components/AuthForm.jsx

```diff
@@ -0,0 +1,68 @@
+import { useState, useEffect } from "react";
+import { auth } from "../firebase";
+import {
+  signInWithEmailAndPassword,
+  createUserWithEmailAndPassword,
+  onAuthStateChanged,
+} from "firebase/auth";
+import { useNavigate } from "react-router-dom";
+
+const AuthForm = (props) => {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+
+  const signIn = () => {
+    signInWithEmailAndPassword(auth, email, password).then((userCredential) => {
+      props.setIsLoggedIn(true);
+      props.setUser(userCredential.user);
+    });
+  };
+
+  const signUp = () => {
+    createUserWithEmailAndPassword(auth, email, password).then(
+      (userCredential) => {
+        props.setIsLoggedIn(true);
+        props.setUser(userCredential.user);
+      }
+    );
+  };
+
```

Comment:
Nice implementation of the authentication system. It could be cleaner, right now you are doubling up your state updates within this component 

Lines 16,17 -- 24/25 
Will update your state

Then in the onAuthStateChanged listener you update state again, remove some redundant code. 

 I do like the user of useNavigate, to move the user to the form after loggin in.